lib/util/makeRequest.ts

```diff
@@ -70,34 +83,47 @@ export type Options = request.CoreOptions &
  */
 export async function makeRequest(
   options: Options,
-  reportError: jsonParser.ReportError,
+  reportError: jsonParser.ReportError
 ): Promise<SwaggerObject> {
-  const promise = new Promise<SwaggerObject>((resolve, reject) => {
-    request(options, (err, response, responseBody) => {
+  return retry(
+    async (bail, retryNr) => {
+      const { err, response, responseBody } = await promisifiedRequest(options)
+
       if (err) {
-        reject(err)
+        const message = `Request to ${
+          options.url
+        } failed with error ${err} on retry number ${retryNr}.`
+
+        throw new Error(message)
       }
+
       if (options.errorOnNon200Response && response.statusCode !== 200) {
         const msg = `StatusCode: "${
           response.statusCode
         }", ResponseBody: "${responseBody}."`
-        reject(new Error(msg))
+
+        return bail(new Error(msg))
       }
+
       let res = responseBody
+
       try {
         if (typeof responseBody.valueOf() === "string") {
           res = parseContent(options.url, responseBody, reportError)
         }
       } catch (error) {
         const url = options.url
         const text = util.inspect(error, { depth: null })
-        const msg = `An error occurred while parsing the file ${url}. The error is:\n ${text}.`
-        const e = new Error(msg)
-        reject(e)
+        const msg = `An error occurred while parsing the file ${url} on
+        retry number ${retryNr}.. The error is:\n ${text}.`
+
+        return bail(new Error(msg))
       }
 
-      resolve(res)
-    })
-  })
-  return await promise
+      return res
+    },
+    {
+      retries: 3
```

Comment:
do you mind specifing the defaults explicitly ? ( for the backoff logic and whatever else )